components/portal/messages/services.js

```diff
@@ -208,6 +208,61 @@ define(['angular'], function(angular) {
         };
 
         /**
+         * Filter the array of messages based on if
+         * a url to get the title was requested
+         * @param {Array} messages An array of message objects
+         * @return {filteredMessages[]} an array of messages that
+         *   includes messages that requested a title from a url
+         */
+        var getMessagesByTitle = function(messages) {
+          var promises = [];
+          var filteredMessages = [];
+          var toBeDiscarded = [];
+          angular.forEach(messages, function(message) {
+            if (message.titleUrl) {
```

Comment:
:art: If possible prefer `Array.prototype.filter` for messages with a `titleUrl` to reduce complexity of `forEach`